scripts/1.TCGA-MLexample.py

```diff
@@ -1,247 +1,250 @@
 
 # coding: utf-8
 
-# In[1]:
-
-get_ipython().magic('matplotlib inline')
+# # Create a logistic regression model to predict TP53 mutation from gene expression data in TCGA
 
-
-# In[2]:
+# In[1]:
 
 import os
 import urllib
-import pandas as pd
 import random
-from statsmodels.robust import scale
-from sklearn import preprocessing
-from sklearn.linear_model import SGDClassifier
-from sklearn.cross_validation import StratifiedKFold, train_test_split
-from sklearn.metrics import roc_auc_score, roc_curve
+import warnings
+
+import pandas as pd
 import matplotlib.pyplot as plt
 import seaborn as sns
+from sklearn import preprocessing, grid_search
+from sklearn.linear_model import SGDClassifier
+from sklearn.cross_validation import train_test_split
+from sklearn.metrics import roc_auc_score, roc_curve
 
 
-# In[3]:
+# In[2]:
 
-random.seed(123)
+get_ipython().magic('matplotlib inline')
+plt.style.use('seaborn-notebook')
 
 
-# In[4]:
+# ## Specify model configuration
 
-# We're going to be building a 'TP53' logistic regression classifier with
-# an elastic net penalty
+# In[3]:
+
+# We're going to be building a 'TP53' classifier 
 GENE = 'TP53'
-CLASSIFIER = 'log'
-PENALTY = 'elasticnet'
 
 
-# In[5]:
+# In[4]:
 
 # Parameter Sweep for Hyperparameters
-alpha_params = [0.05, 0.1, 0.2, 0.3]
-l1_params = [0.05, 0.1, 0.2, 0.3]
+param_grid = {
+    'loss': ['log'],
+    'penalty': ['elasticnet'],
+    'alpha': [10 ** x for x in range(-6, 1)],
+    'l1_ratio': [0, 0.05, 0.1, 0.2, 0.5, 0.8, 0.9, 0.95, 1],
+}
 
 
 # *Here is some [documentation](http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html) regarding the classifier and hyperparameters*
 # 
 # *Here is some [information](https://ghr.nlm.nih.gov/gene/TP53) about TP53*
 
-# # Load Data
+# ## Load Data
 
-# In[6]:
+# In[5]:
 
 if not os.path.exists('data'):
     os.makedirs('data')
 
 
+# In[6]:
+
+url_to_path = {
+    # X matrix
+    'https://ndownloader.figshare.com/files/5514386':
+        os.path.join('data', 'expression.tsv.bz2'),
+    # Y Matrix
+    'https://ndownloader.figshare.com/files/5514389':
+        os.path.join('data', 'mutation-matrix.tsv.bz2'),
+}
+
+for url, path in url_to_path.items():
+    if not os.path.exists(path):
+        urllib.request.urlretrieve(url, path)
+
+
 # In[7]:
 
-# Download X Matrix
-url = 'https://ndownloader.figshare.com/files/5514386'
-if not os.path.exists('data/expression.tsv.bz2'):
-    urllib.request.urlretrieve(url, os.path.join('data', 'expression.tsv.bz2'))
+get_ipython().run_cell_magic('time', '', "path = os.path.join('data', 'expression.tsv.bz2')\nX = pd.read_table(path, index_col=0)")
```

Comment:
See [Cell 7](https://github.com/dhimmel/machine-learning/blob/9557e47cfe4514957247778c10dbc005965c9e5b/1.TCGA-MLexample.ipynb). I used the time magic because I thought it would be helpful to report runtime. The downside is that it doesn't convert nicely to scripts.

FYI, here's the code of cell 6 that created the line you commented on:

```python
%%time
path = os.path.join('data', 'expression.tsv.bz2')
X = pd.read_table(path, index_col=0)
```